packages/framework-core/src/services/read-model-store.ts

```diff
@@ -43,6 +43,10 @@ export class ReadModelStore {
           ' to build new state of read model ${readModelName} with ID ${readModelID}'
         )
         const newReadModel = this.reducerForProjection(projectionMetadata)(entitySnapshot, readModel)
+
+        if (newReadModel === deleteReadModel) {
+          return this.provider.readModels.deleteReadModel(this.config, this.logger, readModelName, readModel)
+        }
```

Comment:
I think that here you could do something like this to avoid the extra type:

```suggestion
        if (newReadModel === null) {
          return this.provider.readModels.deleteReadModel(this.config, this.logger, readModelName, readModel)
        }
```